Git info functionality introduced

dokato opened the pull request:
Introduced option to display current git commit and branch.

przytu1 commented on R/git.R:
```diff
@@ -0,0 +1,22 @@
+#' @import shiny
+#' @export
+git_info <- function() {
+  print(getwd())
```
I don't think we should leave `print` statements like that. It is side effect. Let developer choose, maybe he will read `git_info()` and then prints it somewhere.

przytu1 commented on R/git.R:
```diff
@@ -0,0 +1,22 @@
+#' @export
+git_info <- function() {
+  print(getwd())
+  repo <- init(path = ".", bare = FALSE)
```
This is dangerous. Why do we initialize a git repo?
When I ran the example code in the console (shiny app was not stored in any file), my current directory was initialized as git repo. It should be rather "if my getwd() is a repo, then I display git info. Otherwise display that I am not in the repo"

dokato replied:
Yeah, I confused `init` functionality with `repository`. Fixed.

przytu1 commented on R/git.R:
```diff
@@ -0,0 +1,22 @@
+#' @import shiny
+#' @export
+git_info <- function() {
+  print(getwd())
```
I don't think we should leave print statements like that. It is side effect. Let developer choose, maybe he will read git_info() and then prints it somewhere.

dokato replied:
That's development leftover. Removed.

przytu1 commented on R/git.R:
```diff
@@ -0,0 +1,22 @@
+  if (is.null(branch)) {
+    git_message <- "Empty repo."
+  } else {
+    last_commit <- as.character(commits(repo)[[1]])[[4]]
```
Are you sure `[[1]]` and `[[4]]` will always work?

dokato replied:
`[[1]]` I'd leave as this is just a first commit from the top of the history list. `[[4]]` I changed to field name of the list.

przytu1 commented on R/git.R:
```diff
@@ -0,0 +1,22 @@
+    git_message <- "Empty repo."
+  } else {
+    last_commit <- as.character(commits(repo)[[1]])[[4]]
+    git_message <- span(paste("Branch:", branch),
```
I suggest using `glue()` for concatenating text. You could easier notice that you added different number of spaces.

```
glue("Branch: {branch}{br()}Last commit: {last_commit}")

dokato replied:
You're right. I used `glue` with some html styling so it looks as in the fig below.

dokato commented:
@przytu1 Thanks for helping with this. I introduced your comments and added some HTML styling, so now the git notification looks like this:
![Screenshot 2019-04-30 at 11 40 13](https://user-images.githubusercontent.com/4547289/56956995-aeb84600-6b3d-11e9-99cb-2ccc9cb9cbf3.png)

przytu1 commented on R/git.R:
```diff
@@ -0,0 +1,34 @@
+  if (is.null(repo)) {
+    git_message <- "It looks like you're not in a valid git repo."
+  } else {
+    branch <- repository_head(repo)[[1]]
```
`repository_head` function is available only from 0.22.1 git2r version. Please add in DESCRIPTION:

```
Imports:
    git2r (>= 0.22.1)
```

dokato replied:
Thanks for spotting that. Fixed.
